### PR TITLE
Switch from sin(x) to exp(x) as the target function in polynomial_autograd 

### LINKED ISSUE #3596: Proposal: replace sin(x) with exp(x) in polynomial_autograd tutorial
I have a branch ready that uses `exp(x)` instead of `sin(x)` as the `y` (target) variable. The reason for this is faster and clearer convergence, since `exp(x)` is well approximated by a Taylor expansion around the origin. Also, all the derivatives are `exp(0)=1`, so the polynomial coefficients are trivial (`1/n!` for the `n-th` term). 

I think this is a minor but positive improvement for someone new trying to kick the tires, and experiment with different learning rates, etc. 

Thanks!
Leo

### PR DESCRIPTION
**Pull request by leomilano**
Fixes #3596

## Description
This PR  branch uses 'exp(x)' instead of 'sin(x)' as the 'y' (target) variable. The reason for this is faster and clearer convergence, since exp(x) is well approximated by a Taylor expansion around the origin. Also, all the derivatives are exp(0)=1, so the polynomial coefficients are trivially '1/n! '(for the 'n-th' term).

I am also improving the information output to the user inside the optimization loop.

Finally, I am making the top doc string a raw string, to avoid a python3 warning about it.

## Checklist

- [x] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [x] Only one issue is addressed in this pull request
- [x] Labels from the issue that this PR is fixing are added to this pull request
- [x] No unnecessary issues are included into this pull request.

### CODE REVIEW

**Comment by svekars** [comment]
@leomilano can you please fix lintrunner? Happy to merge after that

**Comment by leomilano** [comment]
@svekars Will do, shortly

**Comment by leomilano** [comment]
> @leomilano can you please fix lintrunner? Happy to merge after that

Hi, @svekars , I just did - the latest commit passes "lintrunner -m" locally - Thanks much!

**Comment by konjac** [comment]
@leomilano @svekars @albanD 

I have some doubts for this change. I think the updated file actually causes inconsistency.

1. The comment line 5 & 6 are still about fitting for sin(x).
2. This file is included in https://github.com/pytorch/tutorials/blame/3469d47af6e14742990210ec35933ebe73fe380c/beginner_source/pytorch_with_examples.rst#L113 . The whole tutorial page is about fitting for sin(x), incrementally from numpy and manual backward to auto grad. Shouldn't we keep them aligned?

Thus this example source should be reverted to sin(x) unless a comprehensive update to the whole tutorial page.

**Comment by leomilano** [comment]
> @leomilano @svekars @albanD
> 
> I have some doubts for this change. I think the updated file actually causes inconsistency.
> 
> 1. The comment line 5 & 6 are still about fitting for sin(x).
> 2. This file is included in https://github.com/pytorch/tutorials/blame/3469d47af6e14742990210ec35933ebe73fe380c/beginner_source/pytorch_with_examples.rst#L113 . The whole tutorial page is about fitting for sin(x), incrementally from numpy and manual backward to auto grad. Shouldn't we keep them aligned?
> 
> Thus this example source should be reverted to sin(x) unless a comprehensive update to the whole tutorial page.

Good points, @konjac . Sorry about your point 1, I didn't notice. I could quickly address that, but I think point 2 is harder to address, I hadn't noticed the disconnect with the rest of the tutorial. 

The other changes in the PR are minor improvements and I see no reason to revert those. How about this proposal for a new PR:

(a) move back to` sin(x)`
(b) add a comment above that line encouraging the developer taking the tutorial to try other functions with faster convergence, such as exp(x), a polynomial in X, etc. 

My motivation for the change was that it wasn't clear to me that this thing was converging, so I made the local change I wound up submitting. 

Anyways, if this sounds like a good idea, please let me know whether I should open a new issue or refer to this (closed) issue (would we need to reopen it?) 

Thanks so much! - Leo

**Comment by konjac** [comment]
It sounds good to me. 

> > @leomilano @svekars @albanD
> > I have some doubts for this change. I think the updated file actually causes inconsistency.
> > 
> > 1. The comment line 5 & 6 are still about fitting for sin(x).
> > 2. This file is included in https://github.com/pytorch/tutorials/blame/3469d47af6e14742990210ec35933ebe73fe380c/beginner_source/pytorch_with_examples.rst#L113 . The whole tutorial page is about fitting for sin(x), incrementally from numpy and manual backward to auto grad. Shouldn't we keep them aligned?
> > 
> > Thus this example source should be reverted to sin(x) unless a comprehensive update to the whole tutorial page.
> 
> Good points, @konjac . Sorry about your point 1, I didn't notice. I could quickly address that, but I think point 2 is harder to address, I hadn't noticed the disconnect with the rest of the tutorial.
> 
> The other changes in the PR are minor improvements and I see no reason to revert those. How about this proposal for a new PR:
> 
> (a) move back to` sin(x)` (b) add a comment above that line encouraging the developer taking the tutorial to try other functions with faster convergence, such as exp(x), a polynomial in X, etc.
> 
> My motivation for the change was that it wasn't clear to me that this thing was converging, so I made the local change I wound up submitting.
> 
> Anyways, if this sounds like a good idea, please let me know whether I should open a new issue or refer to this (closed) issue (would we need to reopen it?)
> 
> Thanks so much! - Leo

Thank you!

The proposal sounds a reasonable fix.
